### PR TITLE
feat(reconnect): replace top banner with animated sidebar overlay

### PR DESCRIPTION
**Pull request by klopez4212**
## Summary

Replace the top yellow `ConnectionBanner` with an animated overlay that keeps the reconnect card accessible regardless of sidebar state.

## Changes

- **Remove `ConnectionBanner.tsx`** — the yellow bar at the top of the main content area
- **Add `RelayConnectionOverlay.tsx`** — when the sidebar is open, the card stays in the sidebar footer; when collapsed, it slides down to a fixed bottom-left position with a motion animation
- **Fix vertical centering** — title + icon are now properly centered in the connected state (no description pushing the title down)
- **Card keeps full width** (284px) and opaque background when floating
- **Update E2E tests** for the new overlay behavior

### Disconnected (sidebar open)
Card in the sidebar footer:

Disconnected state in sidebar

### Disconnected (sidebar collapsed)
Card slides to fixed bottom-left overlay:

Overlay when sidebar collapsed

### Connected success
After reconnecting, shows success state before auto-dismissing:

Connected success state

### CODE REVIEW

File: desktop/src/app/RelayConnectionOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <motion.div
+          animate={{ opacity: 1, y: 0 }}
+          className="pointer-events-none fixed bottom-3 left-3 z-50 w-[284px]"
+          exit={{ opacity: 0, y: 20 }}
+          initial={{ opacity: 0, y: -20 }}
+          key="relay-connection-overlay"
+          transition={{ duration: 0.25, ease: [0.22, 1, 0.36, 1] }}
+        >
+          <div className="pointer-events-auto rounded-xl bg-background shadow-md">
+            <SidebarRelayConnectionCard
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/app/RelayConnectionOverlay.tsx:74]: Fixed in b375725 — the sidebar footer card is now hidden when the sidebar is collapsed (`sidebarOpen` gate), so only the overlay instance is mounted. No more duplicate test IDs or duplicate a11y elements.

File: desktop/src/app/RelayConnectionOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const { open: sidebarOpen } = useSidebar();
+
+  const shouldShow = card.showSidebarRelayConnectionCard && !sidebarOpen;
+
+  return (
+    <AnimatePresence>
+      {shouldShow ? (
+        <motion.div
+          animate={{ opacity: 1, y: 0 }}
+          className="pointer-events-none fixed bottom-3 left-3 z-50 w-[284px]"
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/app/RelayConnectionOverlay.tsx]: Fixed in b375725 — the overlay now accepts hasWorkspaceRail and isHuddleDrawerOpen props. When the workspace rail is present, left offset bumps to 60px (past the 48px rail + padding). When the huddle drawer is open, bottom offset accounts for --buzz-huddle-drawer-height. No more overlap with the rail or huddle bar.

File: desktop/src/app/RelayConnectionOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+export function RelayConnectionOverlay({
+  errorMessage,
+  hasWorkspaceRail,
+  isHuddleDrawerOpen,
+  relayUrl,
+}: RelayConnectionOverlayProps) {
+  const card = useSidebarRelayConnectionCard(errorMessage, relayUrl);
+  const { open: sidebarOpen } = useSidebar();
+
+  const shouldShow = card.showSidebarRelayConnectionCard && !sidebarOpen;
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/app/RelayConnectionOverlay.tsx]: Fixed in aac866f — the overlay now uses isMobile + openMobile from useSidebar() to detect when the mobile sheet is closed, and shows the overlay in that case. The sidebar footer card is only rendered when the sidebar surface is actually visible (desktop open or mobile sheet open).

File: desktop/src/app/RelayConnectionOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * sidebar footer instead (and this overlay is hidden). Offsets itself for
+ * the workspace rail (48px) and huddle drawer when present.
+ */
+export function RelayConnectionOverlay({
+  errorMessage,
+  hasWorkspaceRail,
+  isHuddleDrawerOpen,
+  relayUrl,
+}: RelayConnectionOverlayProps) {
+  const card = useSidebarRelayConnectionCard(errorMessage, relayUrl);
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/app/RelayConnectionOverlay.tsx]: Fixed in aac866f — added 'disconnected' to the card's degraded-state check in useSidebarRelayConnectionCard, so terminal disconnects now surface the reconnect card. Gated with a hasNonUnreachableError exclusion so auth rejections (non-unreachable errors) still suppress the card and use the dedicated error path. Existing E2E test for auth disconnects still passes.

File: desktop/src/app/RelayConnectionOverlay.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/app/RelayConnectionOverlay.tsx]: Fixed in dc4745d — hoisted `useSidebarRelayConnectionCard` to `AppShell` and pass the single hook result down as a prop to both `AppSidebar` and `RelayConnectionOverlay`. Now there's only one `isDismissed` state instance shared between both surfaces. Dismissing in either the sidebar footer or the overlay dismisses everywhere.

File: desktop/src/features/sidebar/ui/AppSidebar.tsx [hunk 6]
```diff
@@ -763,9 +765,10 @@ export function AppSidebar({
 
           <SidebarFooter>
             <AnimatePresence>
-              {sidebarRelayConnectionCard.showSidebarRelayConnectionCard ? (
+              {sidebarRelayConnectionCard.showSidebarRelayConnectionCard &&
+              sidebarOpen ? (
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/features/sidebar/ui/AppSidebar.tsx]: Fixed in dc4745d — the sidebar footer card condition now uses `isMobile ? openMobile : sidebarOpen` instead of just `sidebarOpen`. When the viewport is below `md` and the mobile sheet is open (`openMobile === true`), the footer card renders correctly even if the desktop `open` state is `false`.

File: desktop/src/features/sidebar/ui/useSidebarRelayConnectionCard.ts [hunk 7]
```diff
@@ -68,9 +68,16 @@ export function useSidebarRelayConnectionCard(
...
+  // rather than a network-level relay-unreachable error. In this case, the
+  // disconnected state should NOT trigger the reconnect card — the app shows
+  // a dedicated error path instead.
+  const hasNonUnreachableError =
+    Boolean(errorMessage) && !hasRelayUnreachableError;
   const isRelayConnectionStateDegraded =
     relayConnectionState === "reconnecting" ||
-    relayConnectionState === "stalled";
+    relayConnectionState === "stalled" ||
+    (relayConnectionState === "disconnected" && !hasNonUnreachableError);
```
[the remark this answers is not in the document]
Comment (klopez4212) [reply desktop/src/features/sidebar/ui/useSidebarRelayConnectionCard.ts:80]: Fixed in d02c615 — added a fallback error pill to `RelayConnectionOverlay` that surfaces non-unreachable disconnect errors (e.g. auth rejections) when the sidebar is hidden. It shows the error message with an AlertCircle icon and `role="alert"`, only when the reconnect card isn't already covering the situation. The reconnect card stays focused on network-level relay-unreachable issues.

**Review by wesbillman — APPROVED** [review]
